### PR TITLE
feat: Add ping function

### PR DESCRIPTION
**Pull request by kristofbolyai**
I am not quite happy with the packages.. This is really an edge-case for us currently. But the implementation seems to work greatly :)

### CODE REVIEW

**Comment by magicus** [comment]
As in this "model" not being a "model" of the Wynncraft world?

**Comment by kristofbolyai** [comment]
> As in this "model" not being a "model" of the Wynncraft world?

Yeah.. But it needs world states model, so it cannot really be a manager. Also, I am not sure if ping is a world function..

File: common/src/main/java/com/wynntils/models/ping/PingModel.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    // We are specifically looking for the packet itself, not it's processing. This is why we are using the PacketEvent.
+    @SubscribeEvent
+    public void onCommandSuggestions(PacketEvent.PacketReceivedEvent<ClientboundCommandSuggestionsPacket> event) {
+        if (event.getPacket().getId() != -1) return;
+
+        lastPing = (int) (System.currentTimeMillis() - lastPingSent);
+
+        // We also need to cancel the packet, as it would be unexpected to Minecraft.
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/ping/PingModel.java:49]: Are you sure it is safe to always cancel the event? Is the thinking that we will only get the packetevent if we are connected to a wynncraft server, and if we are, then the "real" command suggestion packet has already been sent by the server to the client, and it is only sent once per connection?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/ping/PingModel.java:49]: No packets with the id of -1 will be sent from the client (it sends from 0...POS_INF).

Comment (magicus) [reply common/src/main/java/com/wynntils/models/ping/PingModel.java:49]: Ah, I see. Please add this as a comment either when sending or receiving the packet.

**Review by magicus — APPROVED** [review]

**Pull request merged**